server/src/app.py

```diff
@@ -46,6 +49,11 @@ async def get_deadlines(response: Response, outline_file: UploadFile = File(...)
     """Returns the extracted dates and info from the uploaded file"""
     return file_handler.handle_file(outline_file, response)
 
+@app.get("/xlsx")
+async def get_xlsx(info: Request):
+    """ Returns the generated XLSX file """
+    semester = await info.json()
+    return xlsx_handler.get_xlsx_file(semester)
```

Comment:
Would you mind changing this method signature to 
```python
@app.post("/xlsx")
async def get_xlsx(semester: List[dict]):
    """ Returns the generated XLSX file """
    return xlsx_handler.get_xlsx_file(semester)
``` 
(you'll have to import List from typing)
After a bit of research I learned that it's best to use the POST method when sending any sort of data to the server. Also, by changing the type of semester to `List[dict]` (dict is basically json in this case) we can test it out easily at http://localhost:8000/docs#/default/get_xlsx_xlsx_post